### ISSUE TITLE
PayPal Checkout not working, having issues with CORS policy as well as a 400 status code on /webapps/hermes/api

### ISSUE THREAD

**Issue by b101** [paypal/paypal-checkout-components#1075]
### Description

I'm runnin Wordpress 5.1.1 with WooCommerce 3.5.7, I've tried switching back to a standard Wordpress Theme, but it didn't help, so the error shouldn't be theme related. 

**The error**   
Basically I cant checkout with PayPal. I complete all the steps a buyer has to, (choose item, add to cart, go to cart, checkout, input personal information, pick PayPal as payment method, accept terms & conditions, continue to the PayPal Checkout)    
As soon as the PayPal Checkout screen is loaded, I can see in the browsers console that a `XMLHttpRequest` has been `blocked by CORS policy`, because `the "Access-Control-Allow-Origin" header contains multiple values '*, *' `    
If I just ignore this and continue with the PayPal User login anyway (type in my passw and hit the login button), I get logged in, can hit the pay now button and also see a `windowload_timeout_setting` object in my browsers console. Now, when I hit the pay now button to complete the checkout, I get a `400` status code on this url `https://www.paypal.com/webapps/hermes/api/checkout/3CW905064P417844F/session/createPayment`. Inside my browsers Network inspector I can see the Response I got from the ressource which says :    

```
{"ack":"contingency","contingency":"PAYER_CANNOT_PAY","errorData":{},"meta":{"calc":"f5d64a441beee","rlog":"ziydCUnr%2BvbMk%2B6S5nJipzibcB%2ByYmMQ9JkxQMmvdXEZkd4ZCyNbYZbzjFKHYJ%2FXNuljQ%2B%2Fh3zmpeA7co0S%2FXw_169fedb0809"},"server":"Yp3YWBvSfcclYbZKjegjSm9O9rUjsaIJ3WRS_O3P_iDT2oq0nNEsGjz3IuxRQDPJx7eD0eujg1zdrCz4ExpwRM3uC640on2v7cthZisgsxfwdFDYAAFxQ6zz6lnRb94cN6OfgIcoFYCcWEZ7iKBN50_JE8WOXeiEhVBlJd7PiicylkmHXgwerN2ngy4V1gYfHdapy9fX9GjIC15nunOALFCUEc0V-gJLRvCLW1U9PpoJvJpGJCzmq3sLNxb3T7_yMG1rvEKTPe1u1q2tkakOOm"}
```
Also, the `windowload_timeout_setting` object which was loaded as soon as the pay now window appeared, contains two variables which are clearly not set and I think this also a problem, maybe its because of the CORS policy problem, maybe its not related. The first one is `this.constructor.caller` and the second one is `this.constructor.arguments`. For both of them I get `TypeError: 'arguments', 'callee', and 'caller' cannot be accessed in this context.`   

From this point on I can only cancel the checkout and go back to the onlinestore. Here are some screenshots (in order):     
ss one:    
<img width="1644" alt="ss-one" src="https://user-images.githubusercontent.com/11729682/55759121-f248f400-5a58-11e9-922e-9cbf8b862418.png">       
ss two:     
<img width="1646" alt="ss-two" src="https://user-images.githubusercontent.com/11729682/55759144-fbd25c00-5a58-11e9-82e4-7e31661b9c59.png">        
ss three:     
<img width="1641" alt="ss-three" src="https://user-images.githubusercontent.com/11729682/55759152-012fa680-5a59-11e9-9a75-0d2dc592ea64.png">     
ss four:       
<img width="1830" alt="ss-four" src="https://user-images.githubusercontent.com/11729682/55759168-07258780-5a59-11e9-8fed-39e53b8f0aea.png">
       
I don't know how to find out the `paypal.version` from my browser console 😕   
The issue is consistently and can be recreated anytime. I don't have a developer.paypal account, so I can't answer if it occurs there as well.  

Here is a link to a product I made for a test purchase, (it is 1 Cent), so you could recreated the issue by trying to buy it. [Link to test item](https://www.printarzan.com/product/zelda/)

### Steps to reproduce

as described in the description, if the standard checkout procedure is being followed you WILL run into the issue, at the very end of the PayPal checkout.

### Affected browsers

- [x] Chrome Version 73.0.3683.86 
- [x] Safari Version 12.1 (14607.1.40.1.4)
- [ ] Firefox **not tested**
- [ ] Edge **not tested**
- [ ] IE **not tested**
- [ ] Chrome Mobile/Tablet **not tested**
- [ ] Safari Mobile/Tablet **not tested**
- [ ] Web View / Safari ViewController **not tested**
- [ ] Other **not tested**

**Comment by edindubai** [comment]
Same issue, same wordpress and Woocommerce versions. Site has 'Let's Encrypt' SSL, no CDN. I have also tried disabling site isolation trial in Chrome. Same issue on Edge (last 3 screenshots). Similarly couldn't pay in Firefox. Same regardless of payment method, ie paypal balance or cards. Works without issue on sandbox. The fix apparently released for the CORS bug in #1070 didn't have any effect as comments after its closure show so it's effectively still an open bug from our side but not sure if it's that which kills the payment.  These screen shots are from today.

![cors](https://user-images.githubusercontent.com/22129924/55779226-f3f4d500-5ab5-11e9-8bd1-75b041bf1bbd.jpg)
![createPayment](https://user-images.githubusercontent.com/22129924/55779239-fbb47980-5ab5-11e9-9cb3-f686b350a672.jpg)
![createPayment2](https://user-images.githubusercontent.com/22129924/55779245-ff480080-5ab5-11e9-8488-cfc026f42310.jpg)
![createPayment3](https://user-images.githubusercontent.com/22129924/55779250-0111c400-5ab6-11e9-9f4d-c980963535a0.jpg)
![edge1](https://user-images.githubusercontent.com/22129924/55780040-112aa300-5ab8-11e9-86db-8adeb25c268c.png)
![edge2](https://user-images.githubusercontent.com/22129924/55780068-20115580-5ab8-11e9-83fa-f88af0085924.png)
![edge3](https://user-images.githubusercontent.com/22129924/55780074-2273af80-5ab8-11e9-912f-06a86c82263d.png)

**Comment by edindubai** [comment]
So the merchant account technical support said the engineers said the CORS issue should be ignored as the payment flow proceeds correctly to the point of clicking pay and it has no effect on it. They said payment failure is most likely an issue with the status/country of the seller account. I did try to do a paypal 'pay for services' transfer within paypal, to the seller account's email I was using which also failed. So I then used another paypal account which I first checked I could send to using the above email method, then upgraded that to business and tried those new API details with the website I had issues with and although the CORS issue showed again, the live payment completed without issue on both credit card and paypal balance tests (using the same buyer account that couldn't pay previously). So it's a seller account issue not a coding one and the cors error is not the payer_cannot_pay cause.
